src/dfm/projects/data_processing/nat/convert_nat_to_dolma_format.py

```diff
@@ -0,0 +1,71 @@
+"""This script converts the raw Netarkivet data to the jsonl.gz format that
+Dolma expects"""
+
+import gzip
+import json
+import os
+from collections.abc import Generator, Iterable
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import polars as pl
+
+NatEntry = dict[str, Any]
+
+
+def lazyframe_from_parquet(path: Path) -> pl.LazyFrame:
+    return pl.scan_parquet(path)
+
+
+def get_nat_paths() -> Generator[Path, None, None]:
+    folder = Path("/work/netarchive")
+    year_subfolders = folder.glob("*.parquet")
+    for year_subfolder in year_subfolders:
+        yield from year_subfolder.glob("*.parquet")
+
+
+def convert_nat_to_standard_format(nat_df: pl.LazyFrame) -> list[NatEntry]:
+    """Convert Netarkivet entries to the standard format:
+        {
+        "id": "...",             # MANDATORY: source-specific identifier
+        "text": "foo",           # MANDATORY: textual content of the document
+        "source": "...",         # MANDATORY: source of the data, such as peS2o, common-crawl, etc.
+        "added": "...",          # OPTIONAL: timestamp ai2 acquired this data
+        "created": "..."         # OPTIONAL: timestamp when orig document was created (best-guess if not available)
+        "metadata": {...}        # OPTIONAL: source-specific metadata
+    }
+        Handles conversion of the timestamp format
```

Comment:
Martin and I have updated it to:
```
{
    "id": "...",             # MANDATORY: source-specific identifier
    "text": "foo",           # MANDATORY: textual content of the document
    "source": "...",         # MANDATORY: source of the data, such as peS2o, common-crawl, etc.
    "added": "...",          # OPTIONAL: timestamp we acquired this data (time file was created), specified as YYYY-MM-DDTHH:MM:SS.TIMEZONE e.g 2021-04-13T12:52:46.000Z
    "created": "..."         # OPTIONAL: timestamp when orig document was created (best-guess if not available), should be specified as a range; "YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE"
    "metadata": {            # OPTIONAL: source-specific metadata
	"sub-source": "...", # OPTIONAL: E.g. "newspaper_ocr"
	...
	}        
}
```

Reply:
Might make sense to add the year as "sub-source" perhaps?